packages/tracing/src/browser/request.ts

```diff
@@ -140,6 +137,23 @@ export function instrumentOutgoingRequests(_options?: Partial<RequestInstrumenta
   }
 }
 
+/**
+ * Creates a function that determines whether to attach tracing headers to a request.
+ * This was extracted from `instrumentOutgoingRequests` to make it easier to test shouldAttachHeaders.
+ * TODO (v8): Remove `tracingOrigins` which should drastically simplify this function.
+ */
+export function makeShouldAttachHeaders(
```

Comment:
M: Is there a reason this needs to be a factory function? Seems like we could just pull the function itself out to the module scope and it'd be simpler. It'd need to take the two options as a parameter, but that's doable.

```js
export function shouldAttachHeaders(
  url: string,
  options: {
    tracingOrigins: Array<string | RegExp> | undefined;
    tracePropagationTargets: Array<string | RegExp> | undefined;
  },
): boolean {
  const { tracingOrigins, tracePropagationTargets } = options;
  if (tracePropagationTargets || tracingOrigins) {
    return stringMatchesSomePattern(url, tracePropagationTargets || tracingOrigins);
  }
  return stringMatchesSomePattern(url, DEFAULT_TRACE_PROPAGATION_TARGETS);
}
```

Then in v8 it can just become

```js
export function shouldAttachHeaders(url: string, tracePropagationTargets: Array<string | RegExp> | undefined): boolean {
  return stringMatchesSomePattern(url, tracePropagationTargets || DEFAULT_TRACE_PROPAGATION_TARGETS);
}
```

What do you think?

(Regardless, I feel like since we're thinking about it now, to save Future Us the brainpower, we might as well just include the v8 version in a comment  below since it's so short.)

Reply:
My reasoning for going with the factory function was that if we just extracted the `shouldAttachHeaders` function, we'd need to add `tracePropagationTargets` and  (until v8 also) `tracingOrigins` as additional parameters to `fetchCallback` and `xhrCallback`. Using the factory function would enable us to keep the function signatures unchanged while getting testability.  

> since we're thinking about it now, to save Future Us the brainpower, we might as well just include the v8 version in a comment below

Definitely, will add this.

Reply:
Mmmmm... good point. I guess at that point my instinct would be to leave it as a regular function at the module scope, and inside of `instrumentOutgoingRequests` do something like

```js
const shouldAttachHeadersWithOptionsBakedIn = url => shouldAttachHeaders(url, options) 
```

That name is terrible, but you get this idea : let the two option values just live in the closure.

Up to you, though. I just sometimes feel like we've gone a little factory-function crazy (myself included sometimes). But at this point I'm down from M to L, so I'll leave it in your hands. 🙂

Reply:
Sounds like a plan, thanks for the suggestion!

Reply:
[d981da0](https://github.com/getsentry/sentry-javascript/pull/6238/commits/d981da001ff8af82c12c249e50ba6450e12689d1)